src/test/admin-server/mutations/CollectionAuthor.server.integration.ts

```diff
@@ -0,0 +1,272 @@
+import { db, server } from '../../../admin/server';
+import config from '../../../config';
+import { clear as clearDb, createAuthorHelper } from '../../helpers';
+import {
+  CreateCollectionAuthorInput,
+  UpdateCollectionAuthorImageUrlInput,
+  UpdateCollectionAuthorInput,
+} from '../../../database/types';
+
+describe('mutations: CollectionAuthor', () => {
+  const createData: CreateCollectionAuthorInput = {
+    name: 'Agatha Christie',
+    slug: 'agatha-christie',
+    bio: faker.lorem.paragraphs(2),
+    imageUrl: faker.image.imageUrl(),
+    active: true,
+  };
+
+  beforeEach(async () => {
+    await clearDb(db);
+  });
+
+  afterAll(async () => {
+    await db.$disconnect();
+  });
+
+  describe('createCollectionAuthor mutation', () => {
+    const CREATE_COLLECTION_AUTHOR = gql`
+      mutation createCollectionAuthor(
+        $name: String!
+        $slug: String
+        $bio: Markdown
+        $imageUrl: Url
+        $active: Boolean
+      ) {
+        createCollectionAuthor(
+          data: {
+            name: $name
+            slug: $slug
+            bio: $bio
+            imageUrl: $imageUrl
+            active: $active
+          }
+        ) {
+          externalId
+          name
+          slug
+          bio
+          imageUrl
+          active
+        }
+      }
+    `;
+
+    it('creates an author with just the name supplied', async () => {
+      const name = 'Ian Fleming';
+
+      const {
+        data: { createCollectionAuthor: author },
+      } = await server.executeOperation({
+        query: CREATE_COLLECTION_AUTHOR,
+        variables: { name },
+      });
+
+      expect(author.externalId).toBeTruthy();
+      expect(author.name).toEqual(name);
+      expect(author.slug).toEqual(slugify(name, config.slugify));
+    });
+
+    it('creates an author with all props supplied', async () => {
+      const {
+        data: { createCollectionAuthor: author },
+      } = await server.executeOperation({
+        query: CREATE_COLLECTION_AUTHOR,
+        variables: createData,
+      });
+
+      expect(author.externalId).toBeTruthy();
+      expect(author.name).toEqual(createData.name);
+      expect(author.slug).toEqual(createData.slug);
+      expect(author.bio).toEqual(createData.bio);
+      expect(author.imageUrl).toEqual(createData.imageUrl);
+      expect(author.active).toEqual(createData.active);
+    });
+
+    it('fails on attempting to create an author with a duplicate slug', async () => {
+      const variables: CreateCollectionAuthorInput = {
+        name: 'James Bond',
+        slug: 'james-bond',
+      };
+
+      // Create one author
+      await server.executeOperation({
+        query: CREATE_COLLECTION_AUTHOR,
+        variables,
+      });
+
+      // Attempt to use the same slug one more time...
+      await expect(
+        server.executeOperation({
+          query: CREATE_COLLECTION_AUTHOR,
+          variables,
+        })
+        // ...without success
+      ).rejects.toThrowError();
```

Comment:
There is something there for sure. Context? We don't seem to be setting any beyond the db client and S3. Could S3 be the culprit on CircleCI? The tests exit cleanly in local development. Tried disabling the playground. 

One last thing left to try would be set up a separate server config for these tests instead of reusing the existing one, and strip absolutely everything out of there except the db client. If we decide to mock data as well, we could get rid of the db client as well there.

Reply:
I got to the bottom of the wrong error issue - once I removed the Sentry plugin from the server config used in these tests, I started getting the expected errors - either custom ones emitted by the resolvers like the one about duplicate slugs, or standard GraphQL ones for missing variables in inputs, etc. I've updated the tests to expect the right things to happen. 

However (!), the tests still hang on CircleCI.

Reply:
Updated the PR description - now stopping & starting Apollo Server after every test - this mops up open connections.